main.go

```diff
@@ -17,10 +17,27 @@ import (
 )
 
 type config struct {
-	ServerAddr string       `mapstructure:"listen" split_words:"true"`
-	LogLevel   logrus.Level `mapstructure:"log_level" split_words:"true"`
+	ServerAddr     string       `mapstructure:"listen" split_words:"true"`
+	LogLevel       logrus.Level `mapstructure:"log_level" split_words:"true"`
+	WithInterfaces bool         `mapstructure:"interfaces" envconfig:"interfaces" default:"true"`
+	TokenPerHost   tokenMap     `mapstructure:"token" envconfig:"token"`
+}
+
+type tokenMap map[string]string
+
+var errInvalidTokenFormat = errors.New("invalid API token format: must be instance=token")
 
-	TokenPerHost map[string]string `mapstructure:"token" envconfig:"-"`
+func (toks *tokenMap) Decode(v string) error {
```

Comment:
so this is only called when reading the field from an env var?

Reply:
Yes, this implements the [`envconfig.Decoder`](https://pkg.go.dev/github.com/kelseyhightower/envconfig#Decoder) interface. I'll add a comment.